intellij-toml/src/main/grammars/TomlParser.bnf

```diff
@@ -48,7 +48,9 @@ NewLineKeyValue ::= <<atNewLine KeyValue>> {
     recoverWhile = EatUntilNextLine_recover
 }
 
-Key ::= BareKey | BASIC_STRING | LITERAL_STRING
+Key ::= NestedKey
+private NestedKey ::= BasicKey '.' NestedKey | BasicKey
```

Comment:
Now `TableHeader` and `ArrayTableHeader` parse incorrectly. You can use "fake" rule:

```
fake Key ::= BasicKey ('.' BasicKey)*
LongKey ::= BasicKey ('.' BasicKey)* { elementType = Key }
ShortKey ::= BasicKey { elementType = Key }
```

and replace `Key` in `TableHeader` and `ArrayTableHeader` with `ShortKey` and `Key` in `KeyValue` with `LongKey`. This should fix tests.

Reply:
@mchernyavsky Thanks, works great :)